libunsf.c

```
@@ -3324,7 +3339,7 @@ UNSF_SYMBOL void unsf_convert_sf_to_gus(UnSF_Options *options) {
                                     if ((sf_num_instrument_indexes * 4 != subchunk.size) ||
                                         (sf_instrument_indexes)) BAD_SF();
 
-                                    sf_instrument_indexes = malloc(sizeof(sfInstBag) * sf_num_instrument_indexes);
+                                    sf_instrument_indexes = (sfInstBag *) malloc(sizeof(sfInstBag) * sf_num_instrument_indexes);
```

Why do we need to cast?  

* It is unnecessary, as void * is automatically and safely promoted to any other pointer type in this case.
* It can hide an error, if you forgot to include <stdlib.h>. This can cause crashes (or, worse, not cause a crash until way later in some totally different part of the code). Consider what happens if pointers and integers are differently sized; then you're hiding a warning by casting and might lose bits of your returned address.
* It adds clutter to the code, casts are not very easy to read (especially if the pointer type is long).
* It makes you repeat yourself, which is generally bad.

Is there some overriding reason why we 'need' to do this?

---

Which gcc?  All the ones I've tried haven't given a warning.

---

Dumbass github markup interpreted '-' as a history marker, so my original response
was possibly unreadable. The casts fixes compilation with g++, i.e. fixes the -Wc++-compat
warnings. Do a "make CC=g++" on your unmodified tree to see the errors.

---

oh! that happens if you try to compile the code with g++, but gcc doesn't. That is what you wrapped the public functions in extern "c".

Interesting.

---

s/WildMIDI/unsf/  :)

In any case, since we know our data types, we include the necessary headers, so I
believe it's good practice to do the casting to cover all bases. (and WildMIDI does it
too, and yeah it was probably me who did that :)

---

No, WildMIDI... since unsf will be used in WildMIDI eventually.

---

>  But I meant that WildMIDI and unsf are both C, so that works in our favour. ;)

Yeah, that's a given